apps/condo/domains/news/access/NewsItem.js

```diff
@@ -26,11 +29,60 @@ async function canReadNewsItems ({ authentication: { item: user } }) {
 
         if (isEmpty(residents)) return false
 
-        // !!!
-        // TODO (DOMA-5526): check by news item access rules
-        //
+        const organizationsIds = uniq(compact(residents.map(resident => get(resident, 'organization'))))
 
-        return false
+        const scopesConditions = {
+            deletedAt: null,
+            OR: flatten(residents.map((resident) => ([
+                {
+                    AND: [
+                        {
+                            newsItem: { organization: { id: get(resident, 'organization') } },
+                        },
+                        {
+                            OR: [
+                                {
+                                    AND: [
+                                        { property_is_null: true },
+                                        { unitType: null },
+                                        { unitName: null },
+                                    ],
+                                },
+                                {
+                                    AND: [
+                                        { property: { id: get(resident, 'property') } },
```

Comment:
you select property of the resident 3 times, better to move it to a variable

Reply:
refactored